File: src/main/java/ru/translater/common/TaskFactory.java
package ru.translater.common;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import ru.translater.collect.CollectTask;
import ru.translater.format.FormatTask;
import ru.translater.generate.GenerateTask;
import ru.translater.translate.TranslateTask;

@RequiredArgsConstructor
@Component
public class TaskFactory {
    private final CollectTask collectTask;
    private final FormatTask formatTask;
    private final GenerateTask generateTask;
    private final TranslateTask translateTask;

    public ExecutionTask getTaskRunner(TaskType taskType) {
        switch (taskType) {
            case COLLECT:
                return collectTask;
            case FORMAT:
                return formatTask;
            case TRANSLATE:
                return translateTask;
            case GENERATE:
                return generateTask;
        }

        throw new IllegalArgumentException("TaskType: " + taskType.name() + " not found!");
    }
}

File: src/main/kotlin/com/translater/common/factory/TaskFactory.kt
package com.translater.common.factory

import com.translater.collect.service.CollectTaskService
import com.translater.common.TaskType
import com.translater.common.model.ExecutionTask
import com.translater.format.service.FormatTaskService
import com.translater.generate.service.GenerateTaskService
import com.translater.translate.service.TranslateTaskService
import org.springframework.stereotype.Component

@Component
class TaskFactory(
    val collectTaskService: CollectTaskService,
    val formatTaskService: FormatTaskService,
    val generateTaskService: GenerateTaskService,
    val translateTaskService: TranslateTaskService
) {
    fun getTaskRunner(taskType: TaskType): ExecutionTask {
        return when (taskType) {
            TaskType.COLLECT -> collectTaskService
            TaskType.FORMAT-> formatTaskService
            TaskType.GENERATE -> generateTaskService
            TaskType.TRANSLATE -> translateTaskService
            else -> collectTaskService
        }
    }
}